src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -718,5 +920,71 @@ def setup_pex_process(request: PexProcess, pex_environment: PexEnvironment) -> P
     )
 
 
+@frozen_after_init
+@dataclass(unsafe_hash=True)
+class VenvPexProcess:
```

Comment:
What is the thinking on this being a distinct type, rather than a mode of `PexProcess`? ditto `VenvPex` vs `Pex` (which it looks like we end up using union typing on anyway).

Reply:
> What is the thinking on this being a distinct type, rather than a mode of `PexProcess`.

A `PexProcess` has a `Pex` which does not have a `venv_pex.pex.argv0` (the wrapper script)... so PexProcess would need to either gain a `venv_pex` field disjoint to its existing `pex`field or else turn that field into a `Union[Pex, VenvPex]`.

> `VenvPex` vs `Pex` (which it looks like we end up using union typing on anyway).

The `Union` typing is only in a test file that parameterizes a test to test both old and new modes and here the difference is stark. A `Pex` has a digest and a path, A `VenvPex` has a digest, 2 Scripts and a dict of optional other asked-for Scripts. They;re totally different.